GameServer.h

```diff
@@ -108,6 +109,7 @@ class GameServer
 	void DoTimedUpdates();
 	std::size_t FindGameInfoClient(const sockaddr_in& from, unsigned int clientRandValue);
 	std::size_t FindGameInfoServer(const sockaddr_in& from, unsigned int serverRandValue);
+	std::size_t FindGameInfo(const sockaddr_in& from, const std::function <bool(const GameInfo&)>& compareFunction);

```

Comment:
Using `std::function` can be quite heavyweight, and can generate substantially more complicated assembly code to do the actual function call. It might not be the most suitable way to de-duplicate the code in these two relatively simple functions.